src/ResourceManagement/CustomerInsights/CustomerInsights.Tests/Tests/AuthorizationPolicyScenarioTests.cs

```diff
@@ -0,0 +1,241 @@
+                                                            },
+                                                    PrimaryKey =
+                                                        Convert.ToBase64String(Encoding.UTF8.GetBytes("primaryTestRead1")),
+                                                    SecondaryKey =
+                                                        Convert.ToBase64String(Encoding.UTF8.GetBytes("secondaryTestRead1"))
+                                                };
+
+                var policyResourceFormat2 = new AuthorizationPolicyResourceFormat
+                                                {
+                                                    Permissions =
+                                                        new List<PermissionTypes?>
+                                                            {
+                                                                PermissionTypes.Read,
+                                                                PermissionTypes.Write,
+                                                                PermissionTypes.Manage
+                                                            },
+                                                    PrimaryKey =
+                                                        Convert.ToBase64String(Encoding.UTF8.GetBytes("primaryTestRead2")),
+                                                    SecondaryKey =
+                                                        Convert.ToBase64String(Encoding.UTF8.GetBytes("secondaryTestRead2"))
+                                                };
+
+                aciClient.AuthorizationPolicies.CreateOrUpdate(
+                    ResourceGroupName,
+                    HubName,
+                    policyName1,
+                    policyResourceFormat1);
+                aciClient.AuthorizationPolicies.CreateOrUpdate(
+                    ResourceGroupName,
+                    HubName,
+                    policyName2,
+                    policyResourceFormat2);
+
+                TestUtilities.Wait(1000);
+
+                var policyList = aciClient.AuthorizationPolicies.ListByHub(ResourceGroupName, HubName);
+
+                Assert.True(policyList.ToList().Count >= 2);
+                Assert.True(
+                    policyList.ToList().Any(policyReturned => policyName1 == policyReturned.PolicyName)
+                    && policyList.ToList().Any(policyReturned => policyName2 == policyReturned.PolicyName));
+            }
+        }
+
+        [Fact]
+        public void RegeneratePrimaryKey()
+        {
+            using (var context = MockContext.Start(this.GetType().FullName))
+            {
+                var aciClient = context.GetServiceClient<CustomerInsightsManagementClient>();
+
+                var policyName = TestUtilities.GenerateName("testPolicy");
+
+                var policyResourceFormat = new AuthorizationPolicyResourceFormat
```

Comment:
@tjlvtao can you wrap this entire policyResoureFormat generation code into a function, rather than repeating the same exact code with different parameters. Makes the entire code file large.
Unless you have a very good reason to repeat the same code in every test?